packages/funding-rate-proposer/src/proposer.js

```diff
@@ -278,12 +275,104 @@ class FundingRateProposer {
         at: "PerpetualProposer#updateFundingRate",
         message: "Skipping proposal because current rate is within allowed margin of error",
         fundingRateIdentifier,
-        proposedRate: offchainFundingRate,
+        proposedRate: pricefeedPrice,
         currentRate: onchainFundingRate,
         allowedError: this.fundingRateErrorPercent
       });
     }
   }
+
+  // If there is a pending funding rate that can be published, AND the current offchain price differs from the pending
+  // rate more than some threshold, than publish the pending rate
+  async _applyFundingRate(contractAddress) {
+    const cachedContract = this.contractCache[contractAddress];
+    const currentFundingRateData = cachedContract.state.currentFundingRateData;
+    const fundingRateIdentifier = this.hexToUtf8(currentFundingRateData.identifier);
+
+    // Get `getSettleableProposals` in which the proposer was this bot account. Return early if there are no
+    // pending proposals.
+    const publishableProposals = this.optimisticOracleClient.getSettleableProposals(this.account).filter(proposal => {
+      return proposal.requester === contractAddress;
+    });
+    if (publishableProposals.length === 0) {
+      this.logger.debug({
+        at: "PerpetualProposer#applyFundingRate",
+        message: "No pending proposals",
+        fundingRateIdentifier
+      });
+      return;
+    }
+    assert(publishableProposals.length === 1, "Should be a maximum of 1 pending proposal for any identifier");
+    const publishableFundingRate = publishableProposals[0].proposedPrice.toString();
+
+    // Compare the proposed rates to the pricefeed's current rate.
+    const pricefeedPrice = await this._getPriceFeedAndPrice(fundingRateIdentifier);
+    if (!pricefeedPrice) return;
+    let shouldUpdateFundingRate = isDeviationOutsideErrorMargin(
+      this.toBN(publishableFundingRate), // ObservedValue
+      this.toBN(pricefeedPrice), // ExpectedValue
+      this.toBN(this.toWei("1")),
+      this.toBN(this.toWei(this.fundingRateErrorPercent.toString()))
+    );
+    if (shouldUpdateFundingRate) {
+      const apply = cachedContract.contract.methods.applyFundingRate();
```

Comment:
nit: this is not a great var name

Reply:
What would you propose changing it to? I feel that `shouldUpdateFundingRate` directly explains what its set to: if the deviation is outside the error threshold, the pending funding rate should be published. Is `shouldPublishFundingRate` better?

Reply:
I was complaining about the `apply` var name. perhaps `applyFundingRateTransaction` or something?